x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/DatafeedManager.java

```diff
@@ -520,7 +520,12 @@ private void runTask(TransportStartDatafeedAction.DatafeedTask task) {
             // a context with sufficient permissions would coincidentally be in force in some single node
             // tests, leading to bugs not caught in CI due to many tests running in single node test clusters.
             try (ThreadContext.StoredContext ignore = threadPool.getThreadContext().stashContext()) {
-                innerRun(runningDatafeedsOnThisNode.get(task.getAllocationId()), task.getDatafeedStartTime(), task.getEndTime());
+                Holder holder = runningDatafeedsOnThisNode.get(task.getAllocationId());
+                if (holder != null) {
+                    innerRun(holder, task.getDatafeedStartTime(), task.getEndTime());
+                } else {
+                    logger.warn("Datafeed [{}] was closed while being opened", task.getDatafeedId());
```

Comment:
We use the terms “started” and “stopped” with datafeeds instead of “opened” and “closed”, so please could you change those two words in this message.